blog/content/post/2020-07-21-secure-app-coding-practices.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Secure Application – Best coding practices
+abstract: "When building an application, we must assume that it will be exposed to hackers at all times, but also to misuse by ordinary recipients. The danger of the first group seems obvious, but what kind of risk are other users?"
+cover: /covers/secure-app.jpg
+author: malwina.mika
+layout: post
+published: true
+date: 2020-07-22
+archives: "2020"
+
+tags:
+  - security
+  - snyk
+  - helmet
+  - hackers
+  - vulnerability
+categories:
+  - Development
+  - App Dev
+  - Security
+---
+
+When building an application, we must assume that it will be exposed to hackers at all times, but also to misuse by ordinary recipients. The danger of the first group seems obvious, but what kind of risk are other users? You must know that attempts to "cheat the system" are quite common, especially in the e-commerce industry. There is a group of users who count, for example, on errors in calculating discounts or the number of elements. You can expect that if you have an online store, there will be at least a few people who will try to use your site in such a way as to find a vulnerability. Therefore, we must be really careful when building an application. We cannot count on everyone to act according to our plan. A secure application is one in which the author assumes the opposite scenario - such that everyone wants to find a vulnerability in our system. Only such an approach, and then appropriate protection, can give us at least a substitute for security.
+
+We will try to approach the topic from a slightly easier side - what are the best practices for writing secure web applications?
+There are many of them, but here we'll cover some very basic rules to keep in mind.
+
+### Hide sensitive data
+
+This may seem obvious to you but the passwords for our accounts are so important that we should protect them as tightly as possible.
+If you make a test app and keep it in your repository you can say it doesn't matter. However, have you really not used this password somewhere? Hackers know very well that most internet users often use the exact same combination of characters. So you can expose yourself to the fact that someone will try to use the login data found, e.g. on Twitter or Google account. An even worse scenario is possible, because what if you don't change your password before the app is released to the world? Even if you make your repository private from then on, there's no guarantee that someone hasn't already saved your data. Such neglect could cost you a lot.
```

Comment:
Access to repository also can be stolen or otherwise obtained - not common, but possible.

Reply:
patronisning: If you make a test app and keep it in your repository you can say it doesn't matter.